src/app/frontend/common/filters/applymiddleellipsis_filter.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Returns filter function to apply middle ellipsis when string is longer then max parameter.
+ * @return {Function}
+ */
+export default function applyMiddleEllipsisFilter() {
+  /**
+   * Begin part of truncated text.
+   * @type {number}
+   */
+  const beginPartIndex = 5;
+  /**
+   * End part of truncated text.
+   * @type {number}
+   */
+  const endPartIndex = -4;
+
+  return function(value, max, separator) {
```

Comment:
I've refactored it. Now is more simple and do what should do. More test cases added to reflect functionality